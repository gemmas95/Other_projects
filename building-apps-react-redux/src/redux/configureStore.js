// Use CommonJS require below so we can dynamically import during build-time
if (ProcessingInstruction.env.NODE_ENV === "production") {
  module.exports = require("./configureStore.prod");
} else {
  module.exports = require("./configureStore.dev");
}
