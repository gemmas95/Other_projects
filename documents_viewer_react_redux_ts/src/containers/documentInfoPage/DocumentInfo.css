.shortInfo {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.documentInfo > p {
  margin: 30px 0px;
  line-height: 1.5em;
}

.documentInfo img {
  max-width: -webkit-fill-available;
  display: flex;
  margin: auto;
}
