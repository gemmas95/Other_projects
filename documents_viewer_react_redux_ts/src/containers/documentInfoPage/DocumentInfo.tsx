import React, { useEffect, useState } from "react";
import { connect } from "react-redux";
import "./DocumentInfo.css";

// Interfaces
import { RouteComponentProps } from "react-router";
import IDocument from "../../models/document.interface";
import { AppState } from "../../redux/reducers";

interface MatchParams {
  documentId: string;
}

interface Props {
  documents: IDocument[];
}

interface Props extends RouteComponentProps<MatchParams> {}

const DocumentInfo: React.FC<Props> = (props) => {
  const [documentId] = useState<string>(props.match.params.documentId);
  const { documents } = props;

  const [document, setDocument] = useState<null | undefined | IDocument>(null);

  useEffect(() => {
    setDocument(
      (function getDocument(documentId: string): IDocument | undefined {
        return documents.find((doc: IDocument) => {
          return doc.id === documentId;
        });
      })(documentId)
    );
  }, [documentId, documents]);

  return (
    <>
      {document && (
        <div className="documentInfo">
          <h1>{document.title}</h1>
          <div className="shortInfo">
            <p>{document.date}</p>
            <p>{document.type}</p>
          </div>
          {document.type !== "simple" && <p>{document.text}</p>}
          {document.type === "advanced" && document.image && (
            <img src={document.image} alt={document.title} />
          )}
        </div>
      )}
    </>
  );
};

function mapStateToProps(state: AppState) {
  return {
    documents: state.documents.data,
  };
}

export default connect(mapStateToProps)(DocumentInfo);
