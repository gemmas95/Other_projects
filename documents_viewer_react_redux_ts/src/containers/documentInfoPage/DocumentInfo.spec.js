import renderer from "react-test-renderer";
import React from "react";
import DocumentInfo from "./DocumentInfo";
import { BrowserRouter } from "react-router-dom";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { cleanup } from "@testing-library/react";

const mockStore = configureStore([]);
describe("DocumentInfo", () => {
  let store;

  let match = {
    params: {
      documentId: "1",
    },
  };
  beforeEach(() => {
    store = mockStore({
      myState: "sample text",
      documents: {
        data: [{ id: "1" }],
      },
    });

    store.dispatch = jest.fn();
  });

  afterEach(() => {
    cleanup();
  });

  it("should match snapshot with params match docId", () => {
    const DocumentInfoTree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <DocumentInfo match={match} />
        </BrowserRouter>
      </Provider>
    );
    expect(DocumentInfoTree.toJSON()).toMatchSnapshot();
  });
});
