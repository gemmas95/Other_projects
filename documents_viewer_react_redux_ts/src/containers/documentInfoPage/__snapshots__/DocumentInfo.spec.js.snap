// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentInfo should match snapshot with params match docId 1`] = `null`;
